src/DssGateSuck.sol

```diff
@@ -0,0 +1,162 @@
+    }
+
+    // --- Integration Access Control ---
+    mapping (address => uint256) public bud;
+    event Kiss(address a);
+    event Diss(address a);
+    function kiss(address _a) external auth {
+        require(_a != address(0), "bud/no-contract-0");
+        require(bud[_a] == 0, "bud/approved");
+        bud[_a] = 1;
+        emit Kiss(_a);
+    }
+    function diss(address _a) external auth {
+        require(bud[_a] == 1, "bud/not-approved");
+        bud[_a] = 0;
+        emit Diss(_a);
+    }
+    modifier toll { require(bud[msg.sender] == 1, "bud/not-authorized"); _; }
+
+    /// maker protocol vat
+    address public immutable vat;
+    /// maker protocol vow
+    address public immutable vow;
+
+    /// draw limit- total amount that can be drawn from vat.suck
+    uint256 public max; // [rad]
+
+    constructor(address vat_, address vow_) public {
+        wards[msg.sender] = 1;
+        emit Rely(msg.sender);
+
+        vat = vat_; // set vat address
+        vow = vow_; // set vow address
+    }
+
+    // --- Events ---
+    event File(bytes32 indexed what, uint256 data);
+    event Draw(address indexed dst, uint256 amount); // log upon draw
+
+    // --- UTILS ---
+    function _max(uint x, uint y) internal pure returns (uint z) {
+        return x <= y ? y : x;
+    }
+
+    // --- Draw Limits ---
+    /// Update draw limit
+    /// @dev Restricted to authorized governance addresses
+    /// @dev Approved total can be updated to both a higher or lower value
+    /// @param what what value are we updating
+    /// @param data what are we updating it to
+    function file(bytes32 what, uint256 data) external auth {
+        if (what == "max") {
+            max = data; // update approved total amount
+            emit File(what, data);
+        }
+    }
+
+    /// Suck to destination
+    /// @dev Restricted to approved integration addresses
+    /// @param dst who are you sucking to
+    /// @param amt dai amount in rad
+    function suck(address dst, uint256 amt) public toll returns (bool) {
+        require(max >= amt, "dss-gate/insufficient-allowance");
+
+        max = max - amt;
+
+        VatLike(vat).suck(address(vow), dst, amt);
+        emit Draw(dst, amt);
+    }
+
+    /// Suck to sender
+    /// @dev Restricted to approved integration addresses (because it calls the public suck)
+    /// @param amt dai amount in rad
+    function suck(uint256 amt) external {
+        suck(msg.sender, amt);
+    }
+
+    /// Repay dai (reverse suck)
+    function blow(uint256 amt) external {
+        VatLike(vat).suck(msg.sender, address(vow), amt);
+    }
+
+    /// Recover ERC-20 Dai
+    /// In case someone sends ERC-20 dai to this contract
+    /// allows joining it into this contract's balance in vat
+    /// @param join the address of the Dai Join adapter
+    /// @param amt the amount of ERC dai to join [wad]
+    function recover(address join, uint256 amt) external {
+        JoinLike(join).join(address(this), amt);
```

Comment:
Just glancing but not sure this contract has a way to exit the tokens later.
Maube berrer to move to the pause-proxy?
Or otherwise supply an authed transfer function.

Reply:
yeah good call. i added a vat move to put them into the surplus buffer. my thinking is that the reason this receives Dai is that someone was trying to payback what they took out, so thought that made sense. let me know what you think